internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -339,6 +347,7 @@ func schemaSpecs() *schema.Schema {
 		Type:     schema.TypeList,
 		MaxItems: 1,
 		Optional: true,
+		Computed: true,
```

Comment:
**TLDR**: defining the attribute as optional/computed aligns with how the API behaves (returns value even if not defined), and avoids generating an non empty plan after creation. 

Why is this change needed now? In our existing implementation we were able to match replication specs with api object using `zoneName` ([matcher](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/ac3f1fde3a80a42855766139221cb6e7c2f38cfb/internal/service/advancedcluster/model_advanced_cluster.go#L495)), so even during initial creation if `id` was not present `zoneName` was used to match each replication spec. This matching is being used to only define the value in state if defined in the config ([reference](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/ac3f1fde3a80a42855766139221cb6e7c2f38cfb/internal/service/advancedcluster/model_advanced_cluster.go#L571)). 
Now with the new API, matching is only done with id because zoneName is not unique for identifying each replication spec. During creation no existing state is present, so no matching is done and default is to populate all values in the state. If left only as optional the state in the initial read is populated with all info from API, and if some of the attributes (`analytics_specs`, `electable_specs`, `electable_specs`) are not defined in the config a non-empty plan is returned.

Reply:
will this be a breaking change?

Reply:
in any case it's good to take a note and add some e2e [migration] test around this